### ISSUE TITLE
Windowed aggregations: calculate daily aggregates based on time zones other than UTC

### ISSUE THREAD

**Issue by rmoff** [confluentinc/ksql#1968]
Per [StackOverflow question](https://stackoverflow.com/questions/52574467/ksql-change-the-time-zone-in-window-tumbling-clause), if calculating aggregates with KSQL at day-level and above (day/week/month/etc) it would be useful to be able to do these based on the local timezone. 

KSQL windows are based on the epoch (number of milliseconds since Jan 01 1970, **UTC**). 

For aggregates at a second/minute/hour level this doesn't really matter, but for an organisation wanting to report on "yesterday" etc then the window would need to include events within the 24 hour time window based on the specified timezone, not necessarily UTC. 

One idea is if KSQL supported a `DATEADD` function, where the event timestamp could be shifted by the required hours to match the local timezone relative to UTC, and then that used as the timestamp for aggregations. However, this feels like a bit of a hack—open to better suggestions for doing this properly.

**Comment by miguno** [comment]
+1 from a user in an offline conversation at a meetup in Munich.

Feedback was: "Have the ability to specify either the desired time zone or a time offset for windowed aggregations"

**Comment by miguno** [comment]
+1 from an offline user conversation, where the goal is to do windowed aggregations where windows start at midnight Eastern time.

**Comment by mjsax** [comment]
This should be addressed in Kafka Streams first IMHO -- the feature request exist there, too. Afterwards, KSQL can expose it to the users.

**Comment by vcrfxia** [comment]
Looks like there's a [workaround](https://stackoverflow.com/questions/54395471/how-to-handle-different-timezone-in-kafka-streams) in Kafka Streams using custom `TimestampExtractor`s so it is an option to support this in KSQL, though it'd be somewhat hacky.

**Comment by mjsax** [comment]
That's super hacky and I would not recommend it. Why not just work on https://issues.apache.org/jira/browse/KAFKA-7911 directly? Also note, that there is https://github.com/confluentinc/kafka-streams-examples/blob/5.3.1-post/src/test/java/io/confluent/examples/streams/window/DailyTimeWindows.java that we could leverage.

**Comment by vcrfxia** [comment]
Thanks for the update @mjsax . I agree those are both much better options.

**Comment by rmoff** [comment]
We should also support `MONTH` and `YEAR` for window sizes - it sounds like these would fall into this same bucket of work?

**Comment by reneveyj** [comment]
This would be greatly needed. For now I'm doing a table not windowed that groups by the date (modified to be like 2023-xx-01) for all dates of a given month. It works, BUT I don't have the control on retention on the ksqlDB side and it grows indefinitely...
